apps/ensnode/src/lib/label-healing.ts

```diff
@@ -0,0 +1,61 @@
+import { decodeLabelhash } from "@ensdomains/ensjs/utils";
+import { Hex } from "viem";
+
+interface ENSRainbow {
+  /**
+   * Attempts to heal the given label. For healed labels,
+   * it also heals the name if provided.
+   */
+  heal<Label extends string, Name extends string | null>(
+    label: Label,
+    name: Name,
+  ): Promise<[Label, Name] | null>;
+}
+
+/**
+ * A factory function to create an instance of ENSRainbow.
+ * @param ensRainbowApiUrl
+ * @returns
+ */
+function createLabelHealing(ensRainbowApiUrl: URL): ENSRainbow {
+  /**
+   * Fetches the healed label for the given unknown label.
+   * @param labelhash a hash of the unknown label
+   * @returns healed label if available
+   */
+  async function fetchHealedLabel(labelhash: Hex) {
+    const response = await fetch(new URL(`/v1/heal/${labelhash}`, ensRainbowApiUrl).toString());
+
+    if (!response.ok) {
+      throw new Error(`Failed to fetch healed label: ${response.statusText}`);
+    }
+
+    return response.text();
+  }
+
+  return {
+    async heal(label, name) {
+      try {
+        const healedLabel = await fetchHealedLabel(decodeLabelhash(label));
```

Comment:
does this need to decodeLabelHash? we should only ever pass in `heal(labelhash: Labelhash)` where Labelhash is a 32byte hex string. i don't think there's a case in which we pass this function an encoded labelhash

```
const label = await labelHealing.heal(labelhash); // Promise<string | null>
```

Reply:
let's also include our `isLabelIndexable` check here, and this service should return null if the label is not indexable.